packages/dds/merge-tree/src/client.ts

```diff
@@ -298,6 +298,10 @@ export class Client {
         this.mergeTree.walkAllSegments(
             this.mergeTree.root,
             (seg) => {
+                assert(
```

Comment:
Do you think this has the potential to be hit? In the past, when we have added new asserts and it has hit, it has become a huge deal in production.

Reply:
assuming we're only calling GC-related methods on the summarizer container, I'd be surprised to see it get hit. Experimental shared-tree has a similar assert in their summarizeCore path that hasn't caused much trouble (I think once we refactored things poorly and it actually did catch an issue before deploying too far). I did think about if we have a reasonable backoff plan though--are we still flighting GC somehow so that we'd have a good mitigation path?

Reply:
Do you mean like disabling GC as mitigation?

Reply:
My concern comes from this work item and the discussion in it - https://dev.azure.com/fluidframework/internal/_workitems/edit/1285?src=WorkItemMention&src-action=artifact_link.

Currently, it's possible that during summarization, ops can be sent since we are loading user code in summarizer. If that's the case, we can never safely say that summarizer won't have any local ops. This means that summarizer client can send ops in theory at least.

Reply:
Ooh, that work item is interesting background. Given that knowledge I definitely agree we shouldn't check in this assert. It would make sense the equivalent assertion in experimental shared-tree never gets hit, since it's app-specific mixins.

I've converted the assert to an error event.

Reply:
sounds good!